hw_8/hw_8.2.js

```diff
@@ -0,0 +1,284 @@
+      },
+      {
+        id: 7,
+        name: "Отдел маркетинга",
+        employees_count: 20,
+      },
+      {
+        id: 8,
+        name: "Отдел охраны труда",
+        employees_count: 5,
+      },
+    ]
+  },
+  {
+    id: 9,
+    name: "Предприятие 3",
+    departments: [
+      {
+        id: 10,
+        name: "Отдел аналитики",
+        employees_count: 0,
+      },
+    ]
+  }
+]
+
+/*
+1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников. Для предприятия посчитать сумму всех сотрудников во всех отделах.
+Пример:
+Предприятие 1 (45 сотрудников)
+- Отдел тестирования (10 сотрудников)
+- Отдел маркетинга (20 сотрудников)
+- Администрация (15 человек)
+Предприятие 2 (75 сотрудников)
+- Отдел разработки (50 сотрудников)
+- Отдел маркетинга (20 сотрудников)
+- Отдел охраны труда (5 сотрудников)
+Предприятие 3 (нет сотрудников)
+- Отдел аналитики (нет сотрудников)
+*/ 
+
+const getEnterpriseInfo = companyArr => {
+	companyArr.forEach(comp => {
+		console.log(`${comp.name} (${countCompanyEmployees(comp.departments)} сотрудников)`);
+		comp.departments.forEach(depart => {
+			console.log(`- ${depart.name} (${depart.employees_count} сотрудников)`);
+		})
+	});
+}
+
+const countCompanyEmployees = depart => depart.reduce((total, dep) => 
+	total + dep.employees_count, 0);
+
+getEnterpriseInfo(enterprises);
+
+
+/*
+2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+Пример:
+getEnterpriseName(4) // Предприятие 1
+getEnterpriseName("Отдел маркетинга") // Предприятие 2
+*/
+
+function getCompanyNameByDepartId(identifier) {
+	let compName; 
+	enterprises.find(comp => 
+		comp.departments.find(depart => 
+			depart.id === identifier ? compName = comp.name : false))
+	return compName;
+}
+
+
+function getCompanyNameByDepartName(identifier) {
+	let compName; 
+	enterprises.find(comp => 
+		comp.departments.find(depart =>
+			depart.name === identifier ? compName = comp.name : false))
+	return compName;
+}
+
+
+const getCompanyName = identifier => {
+	if (getCompanyNameByDepartId(identifier)) {
+		return `This identifier is valid for ${getCompanyNameByDepartId(identifier)}.`
+	} else if (getCompanyNameByDepartName(identifier)) {
+		return `This identifier is valid for ${getCompanyNameByDepartName(identifier)}.`
+	} else return 'Enter valid data.'
+}
+
+console.log(getCompanyName('Отдел маркетинга'));
+console.log(getCompanyName('Отдел электроники'));
+console.log(getCompanyName(7));
+console.log(getCompanyName(9));
+
+
+/*
+3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+Пример:
+addEnterprise("Название нового предприятия")
+*/
+
+function getMaxId(companyArr) {
+	const idArr = [];
+	companyArr.forEach(comp => {
+		idArr.push(comp.id);
+		comp.departments.forEach(depart => idArr.push(depart.id));
+	});
+	const maxId = Math.max(...idArr);
+	return maxId;
+}
```

Comment:
In all places in code you are adding +1 to the result of getMaxId. Maybe better to call the function getNewId and return getMaxId + 1?)